sdk/storage/inc/datalake/directory_client.hpp

```diff
@@ -0,0 +1,157 @@
+     */
+    std::string GetUri() const { return m_blobClient.GetUri(); }
+
+    /**
+     * @brief Gets the directory's primary uri endpoint. This is the endpoint used for dfs
+     * endpoint only operations
+     *
+     * @return The directory's primary uri endpoint.
+     */
+    std::string GetDfsUri() const { return m_dfsUri.ToString(); }
+
+    // Feature not yet enabled for 2019-12-12.
+    ///**
+    // * @brief Sets POSIX access control rights on files and directories under given directory
+    // *        recursively.
+    // * @param mode Mode PathSetAccessControlRecursiveMode::Set sets POSIX access control rights on
+    // *             files and directories, PathSetAccessControlRecursiveMode::Modify modifies one
+    // or
+    // *             more POSIX access control rights  that pre-exist on files and directories,
+    // *             PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access
+    // *             control rights that were present earlier on files and directories
+    // * @param acls Sets POSIX access control rights on files and directories. Each access control
+    // *             entry (ACE) consists of a scope, a type, a user or group identifier, and
+    // *             permissions.
+    // * @param options Optional parameters to set an access control recursively to the resource the
+    // *                directory points to.
+    // * @return Azure::Core::Response<DirectorySetAccessControlRecursiveInfo>
+    // * @remark This request is sent to dfs endpoint.
+    // */
+    // Azure::Core::Response<DirectorySetAccessControlRecursiveInfo> SetAccessControlRecursive(
+    //    PathSetAccessControlRecursiveMode mode,
+    //    std::vector<Acl> acls,
+    //    const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions())
+    //    const;
+
+    /**
+     * @brief Create a directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to create the directory the path points to.
+     * @return Azure::Core::Response<DirectoryInfo>
+     * @remark This request is sent to dfs endpoint.
+     */
+    Azure::Core::Response<DirectoryInfo> Create(
+        const DirectoryCreateOptions& options = DirectoryCreateOptions()) const
+    {
+      return PathClient::Create(PathResourceType::Directory, options);
+    }
+
+    /**
+     * @brief Renames a directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param destinationDirectoryPath The destinationPath this current directory is renaming to.
+     * @param options Optional parameters to rename a resource to the resource the destination
+     * directory points to.
+     * @return Azure::Core::Response<DirectoryRenameInfo>
+     * @remark This will change the URL the client is pointing to.
```

Comment:
Does this align with how other sdk are doing?

Reply:
Because this makes the whole client non-thread-safe.

Reply:
Yes, it is what .NET does.

Reply:
No, it's not